j5/backends/console/arduino/uno.py

```diff
@@ -0,0 +1,113 @@
+        self.mode = mode
+        self.digital_state = digital_state
+
+
+class ArduinoUnoConsoleBackend(GPIOPinInterface, LEDInterface, Backend):
+    """Console Backend for the Arduino Uno."""
+
+    environment = ConsoleEnvironment
+    board = ArduinoUnoBoard
+
+    @classmethod
+    def discover(cls) -> Set[Board]:
+        """Discover boards that this backend can control."""
+        raise NotImplementedError("The Console Backend cannot discover boards.")
+
+    def __init__(self, serial: str, console_class: Type[Console] = Console) -> None:
+        self._serial = serial
+
+        self._pins: Mapping[int, PinData] = {
+            i: PinData(mode=GPIOPinMode.DIGITAL_OUTPUT, digital_state=False)
+            for i in range(2, 20)
+            # Digital 2 - 13
+            # Analogue 14 - 19
+        }
+
+        # Setup console helper
+        self._console = console_class(f"{self.board.__name__}({self._serial})")
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """The firmware version reported by the board."""
+        return None  # Console, so no firmware
+
+    def set_gpio_pin_mode(self, identifier: int, pin_mode: GPIOPinMode) -> None:
+        """Set the hardware mode of a GPIO pin."""
+        self._console.info(f"Set pin {identifier} to {pin_mode.name}")
+        self._pins[identifier].mode = pin_mode
+
+    def get_gpio_pin_mode(self, identifier: int) -> GPIOPinMode:
+        """Get the hardware mode of a GPIO pin."""
+        return self._pins[identifier].mode
+
+    def write_gpio_pin_digital_state(self, identifier: int, state: bool) -> None:
+        """Write to the digital state of a GPIO pin."""
+        if self._pins[identifier].mode is not GPIOPinMode.DIGITAL_OUTPUT:
+            raise ValueError(f"Pin {identifier} mode needs to be DIGITAL_OUTPUT"
+                             f"in order to set the digital state.")
+        self._console.info(f"Set pin {identifier} state to {state}")
+        self._pins[identifier].digital_state = state
+
+    def get_gpio_pin_digital_state(self, identifier: int) -> bool:
+        """Get the last written state of the GPIO pin."""
+        if self._pins[identifier].mode is not GPIOPinMode.DIGITAL_OUTPUT:
+            raise ValueError(f"Pin {identifier} mode needs to be DIGITAL_OUTPUT"
+                             f"in order to read the digital state.")
+        return self._pins[identifier].digital_state
+
+    def read_gpio_pin_digital_state(self, identifier: int) -> bool:
+        """Read the digital state of the GPIO pin."""
+        if self._pins[identifier].mode not in [
+            GPIOPinMode.DIGITAL_INPUT_PULLUP,
+            GPIOPinMode.DIGITAL_INPUT,
+            GPIOPinMode.DIGITAL_INPUT_PULLDOWN,
+        ]:
+            raise ValueError(f"Pin {identifier} mode needs to be DIGITAL_INPUT_*"
+                             f"in order to read the digital state.")
+        return self._console.read(f"Pin {identifier} digital state [bool]", bool)
```

Comment:
Consistency with https://github.com/j5api/j5/blob/290fdbc/j5/backends/console/sr/v4/power_board.py#L110.

```suggestion
        return self._console.read(f"Pin {identifier} digital state [true/false]", bool)
```